src/js/samples/components/Item/EndIcon.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import styled from "styled-components";
+import { Icon, Loader } from "../../../base";
+
+const SampleIconContainer = styled.div`
+    align-items: center;
+    background: none;
+    bottom: 0;
+    display: flex;
+    justify-content: center;
+    margin-left: auto;
+    strong {
+        margin-left: 5px;
+    }
+`;
+
+/**
+ * Icon indicating the status of sample
+ *
+ * @param ready - Whether the sample is ready
+ * @param onClick - Callback to handle click event
+ */
+export function EndIcon({ ready, onClick }) {
```

Comment:
Add a TS interface for the props on this component.

```suggestion
export function SampleItemEndIcon({ ready, onClick }) {
```